File: src/examples/golang/orders/create/create_order.go
package main

import (
	"encoding/json"
	"fmt"
	"os"

	"github.com/EasyPost/easypost-go/v2"
)

func main() {
	apiKey := os.Getenv("EASYPOST_TEST_API_KEY")

	client := easypost.New(apiKey)

	// Create an order containing multiple shipments
	order, err := client.CreateOrder(
		&easypost.Order{
			ToAddress:   &easypost.Address{ID: "adr_123..."},
			FromAddress: &easypost.Address{ID: "adr_123..."},
			Shipments: []*easypost.Shipment{
				{
					Parcel: &easypost.Parcel{
						// PredefinedPackage: "FedExBox",
						Weight: 10.2,
					},
				},
				{
					Parcel: &easypost.Parcel{
						// PredefinedPackage: "FedExBox",
						Weight: 17.5,
					},
				},
			},
		},
		&easypost.CarrierAccount{ID: "ca_123..."},
	)
	if err != nil {
		fmt.Fprintln(os.Stderr, "error creating order:", err)
		os.Exit(1)
	}

	prettyJSON, err := json.MarshalIndent(order, "", "    ")
	if err != nil {
		fmt.Fprintln(os.Stderr, "error creating JSON:", err)
	}
	fmt.Printf("%s\n", string(prettyJSON))
}

File: src/examples/php/orders/create-order.php
<?php

require_once '/Users/jhammond/git/easypost/easypost-tools/vendor/autoload.php';

use Dotenv\Dotenv;
use EasyPost\Address;
use EasyPost\EasyPost;
use EasyPost\Error;
use EasyPost\Order;

$dotenv = Dotenv::createImmutable('/Users/jhammond/git/easypost/easypost-tools');
$dotenv->load();
EasyPost::setApiKey($_ENV['EASYPOST_TEST_API_KEY']);

$to_address = Address::create(
    [
        "street1" => "417 MONTGOMERY ST",
        "street2" => "FL 5",
        "city"    => "San Francisco",
        "state"   => "CA",
        "zip"     => "94104",
        "country" => "US",
        "company" => "EasyPost",
        "phone"   => "415-123-4567"
    ]
);

$from_address = Address::create(
    [
        "street1" => "417 MONTGOMERY ST",
        "street2" => "FL 5",
        "city"    => "San Francisco",
        "state"   => "CA",
        "zip"     => "94104",
        "country" => "US",
        "company" => "EasyPost",
        "phone"   => "415-123-4567"
    ]
);

try {
    $order = Order::create([
        "to_address" => $to_address,
        "from_address" => $from_address,
        "shipments" => [
            [
                "parcel" => [
                    "predefined_package" => "FedExBox",
                    "weight" => 10.2
                ]
            ],
            [
                "parcel" => [
                    "predefined_package" => "FedExBox",
                    "weight" => 17.5
                ]
            ],
        ],
        "carrier_accounts" => ["ca_123..."],
        "service" => "FEDEX_GROUND",
        "carrier" => "FedEx",
    ]);
    echo $order;
} catch (Error $exception) {
    echo $exception;
}
